Misc/NEWS.d/next/Core and Builtins/2022-03-22-15-12-28.bpo-42197.SwrrFO.rst

```diff
@@ -0,0 +1,2 @@
+:c:func:`PyFrame_FastToLocalsWithError` and :c:func:`PyFrame_LocalsToFast` are no longer
+called during profiling nor tracing. C code can access the ``f_locals`` attribute of :c:type:`PyFrameObject` by calling :c:func:`PyFrame_GetLocals`.
```

Comment:
Please copy this paragraph to What's New In Python 3.11 > C API > Porting to Python 3.11:
https://docs.python.org/dev/whatsnew/3.11.html#id2

By the way, in 2013, it was decided to **not** document these functions: https://bugs.python.org/issue19431

Maybe we should revisit this decision? What do you think? I can document them in the new frame.rst doc if you want.

Reply:
There should be no need to document this is the porting to 3.11 section. It should "just work".
The important change C code must make is to call `PyFrame_GetLocals()` instead of access the `f_locals` field directly. That is already mentioned here and in the whatsnew file.

Reply:
Developers of profilers and debuggers (implemented in C) will have to dig into Changelog or even the Python bug tracker to understand what's going on. Since it's a behavior change, it should be documented.

I agree that profilers and debuggers implemented in pure Python will not be affected. And people not using profiling or debugging will not be affected :-)

Reply:
OK, I've added a section to the whatsnew